src/material/tabs/tab.ts

```diff
@@ -95,6 +95,9 @@ export class MatTab extends _MatTabMixinBase implements OnInit, CanDisable, OnCh
   get content(): TemplatePortal | null {
     return this._contentPortal;
   }
+  set content(contentPortal: TemplatePortal | null) {
+    this._contentPortal = contentPortal;
```

Comment:
Also this approach feels like it's prone to race conditions. If you look below in `ngOnInit`, the `_contentPortal` still gets re-assigned so if you were to assign it before `ngOnInit` (e.g. in the constructor), your custom portal will be replaced.